pstoreds/protobook.go

```diff
@@ -0,0 +1,118 @@
+package pstoreds
+
+import (
+	"fmt"
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+)
+
+type dsProtoBook struct {
+	lks  [256]sync.RWMutex
+	meta pstore.PeerMetadata
+}
+
+var _ pstore.ProtoBook = (*dsProtoBook)(nil)
+
+func NewProtoBook(meta pstore.PeerMetadata) pstore.ProtoBook {
+	return &dsProtoBook{meta: meta}
+}
+
+func (pb *dsProtoBook) lock(p peer.ID) {
+	pb.lks[byte(p[len(p)-1])].Lock()
```

Comment:
nit: a `segment(p peer.ID) *segment` function would be nice (could even drop all of these functions and just call `pb.segment(p).Lock()`.

Reply:
ok, will do. I'll call it `get` similar to how we do it in the memory peerstore.

Reply:
done.